src/app/api/notification/notification.service.ts

```diff
@@ -139,6 +140,42 @@ export class NotificationService extends BaseService {
     // )
   }
 
+  async readAllUserNotifications(userId: string, userType: AssigneeType) {
+    const copilot = new CopilotAPI(this.user.token)
+
+    if (userType === AssigneeType.internalUser) {
+      return
+    }
+    if (userType === AssigneeType.client) {
+      return await this.markAllAsReadForClients([userId])
+    }
+    if (userType === AssigneeType.company) {
+      // In this case, we need to find out all the clients inside this deleted company, then pass that array to markAllAsReadForClients
+      const clients = await copilot.getCompanyClients(userId)
+      return await this.markAllAsReadForClients(clients.map((client) => client.id))
+    }
+
+    throw new APIError(httpStatus.INTERNAL_SERVER_ERROR, 'Failed to parse assignee')
+  }
+
+  async markAllAsReadForClients(clientIds: string[]) {
```

Comment:
Since we have used this in a single method above, this should better be defined as `private`

Also, copilot instance can be passed as a param?

Reply:
I actually removed the `private` identifier on this one since this method is very handy for a number of use cases - I want it to be accessible as a public method.

Reply:
In that case, passing `copilot` as param would be counterintuitive as well :bow: